### PR TITLE
[FLINK-22795] Throw better exception when executing remote SQL file in SQL Client

### PR DESCRIPTION
**Pull request by JasonLeeCoding**
## What is the purpose of the change

*Throw better exception when executing remote SQL file in SQL Client*

## Brief change log

*add checkArgs method in SqlClient.java*

## Verifying this change

This change added tests and can be verified as follows:

*add testExecuteSqlWithHDFSFile in SqlClientTest.java*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by JasonLeeCoding** [comment]
@fsk119 CI failed ,but I didn't find the reason, please take a look , thanks.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java [hunk 1]
```diff
@@ -213,6 +215,20 @@ protected static void startClient(String[] args, Supplier<Terminal> terminalFact
         }
     }
 
+    private static void checkArgs(String[] args) {
+        for (String arg : args) {
+            checkFilePath(arg);
```
Comment (fsk119) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:220]: Why every arg needs to check path.. Only the values of the optoin `-i` and `-f` need check.

**Review by fsk119 — COMMENTED** [review]
Do you reopen a new PR? It loses our discussion before.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java (hunk 1 above)
Comment (JasonLeeCoding) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:220]: yes ,I closed the previous PR, and I will fix it. only check -i -f.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java [hunk 2]
```diff
@@ -179,6 +179,8 @@ protected static void startClient(String[] args, Supplier<Terminal> terminalFact
         switch (mode) {
             case MODE_EMBEDDED:
                 final CliOptions options = CliOptionsParser.parseEmbeddedModeClient(modeArgs);
+                // check arg e.g -i -f
+                checkArgs(args);
```
Comment (Airblader) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:183]: In the line above the args are already parsed into `CliOptions`. Wouldn't it be easier and better to do the validation on that object so we can avoid parsing the arguments (i.e. "-i", "-f") again in `checkArgs`?

Comment (JasonLeeCoding) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:183]: yeah , u are right , This is actually repeated  , I will check it in CliOptionsParser

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**